### PR TITLE
Bug fix dirs 1.1

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
This should go also in here.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
I will add some tests to cover that, tag and merge into master later.

**Comment by ElectricMaxxx** [comment]
I wonder, why the bug does not occur on the frontend tests?

File: Cache/FileCache.php [hunk 1]
```diff
@@ -29,19 +29,18 @@ class FileCache implements CacheInterface, CacheWarmerInterface, CacheClearerInt
...
-        $this->dir = rtrim($dir, '\\/');
+        $this->dir = $baseDir.DIRECTORY_SEPARATOR.rtrim($dir, '\\/');
 
         $this->umask = $umask;
 
-        $dir = $baseDir.DIRECTORY_SEPARATOR.$dir;
-        if (!is_dir($dir)) {
-            mkdir($dir, 0777 & ~$this->umask, true);
+        if (!is_dir($this->dir)) {
+            mkdir($this->dir, 0777 & ~$this->umask, true);
```
Comment (ElectricMaxxx) [inline Cache/FileCache.php]: @lsmith77 btw: is good to have 777 for the file cache stuff? thought 775 for app/cache etc is enough?

Comment (lsmith77) [inline Cache/FileCache.php]: > On 03 Sep 2015, at 12:21, Maximilian Berghoff <notifications@github.com> wrote:
> 
> In Cache/FileCache.php:
> 
> >  
> >          $this->umask = $umask;
> >  
> > -        $dir = $baseDir.DIRECTORY_SEPARATOR.$dir;
> > -        if (!is_dir($dir)) {
> > -            mkdir($dir, 0777 & ~$this->umask, true);
> > +        if (!is_dir($this->dir)) {
> > +            mkdir($this->dir, 0777 & ~$this->umask, true);
> @lsmith77 btw: is good to have 777 for the file cache stuff? thought 775 for app/cache etc is enough?
> 
this is what the umask 0002 is for ..

Comment (ElectricMaxxx) [inline Cache/FileCache.php]: OK, then lets merge this one here, it just adds some tests only.

**Comment by ElectricMaxxx** [comment]
@WouterJ you have got time to merge? I will tag and merge the diff into master.
(this one is the cherry picked fix, that is gone into master today with an extra commit to test it)

**Comment by lsmith77** [comment]
ok. merge 1.1 into master to get the test there as well

**Pull request merged**